src/codewhisperer/service/inlineCompletionService.ts

```diff
@@ -18,13 +18,15 @@ import { getLogger } from '../../shared/logger/logger'
 import { TelemetryHelper } from '../util/telemetryHelper'
 import { runtimeLanguageContext } from '../util/runtimeLanguageContext'
 import { Commands } from '../../shared/vscode/commands2'
-import { getPrefixSuffixOverlap } from '../util/commonUtil'
+import { getPrefixSuffixOverlap, isVscHavingRegressionInlineCompletionApi } from '../util/commonUtil'
 import globals from '../../shared/extensionGlobals'
 import { AuthUtil } from '../util/authUtil'
 import { shared } from '../../shared/utilities/functionUtils'
 import { ImportAdderProvider } from './importAdderProvider'
+import { Mutex } from 'async-mutex'
 
 const performance = globalThis.performance ?? require('perf_hooks').performance
+const mutex = new Mutex()
```

Comment:
Use `async-lock` instead. Adding a new dependency is unnecessary.

Reply:
ACK. I will update it in the next commit.

Reply:
I have switched to async-lock instead.